substrate/primitives/genesis-builder/src/lib.rs

```diff
@@ -17,37 +17,55 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
-//! Substrate genesis config builder
+//! # Substrate genesis config builder.
 //!
-//! For FRAME based runtimes, this runtime interface provides means to interact with
-//! `RuntimeGenesisConfig`. Runtime provides a default `RuntimeGenesisConfig` structure in a form of
-//! the JSON blob.
+//! This crate contains [`GenesisBuilder`], a runtime-api to be implemented by runtimes, in order to
+//! express their genesis state.
 //!
-//! For non-FRAME runtimes this interface is intended to build genesis state of the runtime basing
-//! on some input arbitrary bytes array. This documentation uses term `RuntimeGenesisConfig`, which
-//! for non-FRAME runtimes may be understood as the runtime-side entity representing initial runtime
-//! configuration. The representation of the preset is an arbitrary `Vec<u8>` and does not
-//! necessarily have to represent a JSON blob.
+//! The overall flow of the methods in [`GenesisBuilder`] is as follows:
+//!
+//! 1. [`GenesisBuilder::preset_names`]: A runtime exposes a number of different
+//!    `RuntimeGenesisConfig` variations, each of which is called a `preset`, and is identified by a
+//!    [`PresetId`]. All runtimes are encouraged to expose at least [`DEV_RUNTIME_PRESET`] and
+//!    [`LOCAL_TESTNET_RUNTIME_PRESET`] presets for consistency.
+//! 2. [`GenesisBuilder::get_preset`]: Given a `PresetId`, this the runtime returns the JSON blob
+//!    representation of the `RuntimeGenesisConfig` for that preset. This JSON blob is often mixed
+//!    into the broader `chain_spec`. If `None` is given, [`GenesisBuilder::get_preset`] provides a
```

Comment:
nit: linebreak would be nice here (probably needs reformat):
```suggestion
//!    into the broader `chain_spec`. 
//!    If `None` is given, [`GenesisBuilder::get_preset`] provides a
```